IDR/production/0.collect_screen_metadata.py

```diff
@@ -0,0 +1,81 @@
+from asyncio import subprocess
+import pathlib
+import sys
+import requests
+import subprocess
+import pandas as pd
+from utils.args import *
+
+
+def extract_study_info(session, screen_id):
+    """Pull metadata info per screen, given screen id
+
+    Parameters
+    ----------
+    session: Requests.session()
+        Requests session providing access to IDR API
+    screen_id: int
+        ID of the screen data set
+
+    Returns
+    -------
+    pandas.DataFrame() of metadata per screen id
+    """
+    base_url = "https://idr.openmicroscopy.org/webclient/api/annotations/"
+    screen_url = f"?type=map&screen={screen_id}"
+
+    url = f"{base_url}{screen_url}"
+    response = session.get(url).json()
+
+    annotations = response["annotations"]
+    study_index = [x["ns"] for x in annotations].index(
+        "idr.openmicroscopy.org/study/info"
+    )
+
+    id_ = annotations[study_index]["id"]
+    date_ = annotations[study_index]["date"]
+    name_ = annotations[study_index]["link"]["parent"]["name"]
+
+    details_ = pd.DataFrame(
+        {x[0]: x[1] for x in annotations[study_index]["values"]}, index=[0]
+    ).assign(internal_id=id_, upload_date=date_, idr_name=name_, screen_id=screen_id)
+
+    return details_
+
+
+if __name__ == "__main__":
+    # Define arguments
+    args = collect_screen_metadata_parser().parse_args(sys.argv[1:])
+
+    # Specify where to extract metadata from
+    file_type = args.metadata_fileType
+
+    if file_type == "downloaded_json":
+        ####### Run get_idr_ids.py first
```

Comment:
If `get_idr_ids.py` is a required first step, should it be called `0.get_idr_ids.py` and this file be called `1.collect_screen_metadata.py`?

Reply:
I consider it part of the metadata collection process and since I already have a basic "workflow" with that description, I think it belongs in this file. It would help if I actually added the `subprocess` command though!

Added in 2878c42d8f9b421e8521a431429b2dee5b2cc92f.